app/models/company_value.rb

```diff
@@ -1,3 +1,5 @@
 class CompanyValue < ApplicationRecord
+  has_many :kudos, foreign_key: 'company_value', inverse_of: :company_value, dependent: :restrict_with_error
```

Comment:
You do not have to specify all those parameters, if the name of the associations are matching names of the classes :)
```suggestion
  has_many :kudos, dependent: :restrict_with_error
```

Reply:
I understand ;) Thanks.